src/credentials/FabricTable.cpp

```diff
@@ -994,9 +1014,40 @@ CHIP_ERROR FabricTable::Init(const FabricTable::InitParams & initParams)
         TLV::ContiguousBufferTLVReader reader;
         reader.Init(buf, size);
 
-        ReturnErrorOnFailure(ReadFabricInfo(reader));
+        // TODO: A safer way would be to just clean-up the entire fabric table on this situation...
+        err = ReadFabricInfo(reader);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(FabricProvisioning, "Error loading fabric table: %" CHIP_ERROR_FORMAT ", we are in a bad state!",
+                         err.Format());
+        }
+
+        ReturnErrorOnFailure(err);
     }
 
+    CommitMarker commitMarker;
+    err = GetCommitMarker(commitMarker);
+    if (err == CHIP_NO_ERROR)
+    {
+        // Found a commit marker! We need to possibly delete a loaded fabric
+        ChipLogError(FabricProvisioning, "Found a FabricTable aborted commit for index 0x%x (isAddition: %d), removing!",
+                     static_cast<unsigned>(commitMarker.fabricIndex), static_cast<int>(commitMarker.isAddition));
+
+        mDeletedFabricIndexFromInit = commitMarker.fabricIndex;
```

Comment:
Would it make sense to just return this via an outparam from Init() instead of adding storage for it in perpetuity?  Followup fine for this part if we decide to make that change.

Reply:
I am trying to reduce the massive amount of API changes done here... That's why it's done like this